views/jira-manual-app-creation.hbs

```diff
@@ -0,0 +1,169 @@
+          <input class="text full-width-field" type="text"
+                 data-aui-validation-field required
+                 id="client-id" name="client-id" />
+        </div>
+
+        <div class="jiraManualAppCreation__formTitle">URLs</div>
+        <div class="jiraManualAppCreation__formSubTitle">
+          Copy and paste the following URLs into your GitHub Enterprise Server configuration. This is unique to your site.
+        </div>
+
+        <div class="field-group">
+          <label for="callback-url">Callback URL</label>
+          <input class="text full-width-field" type="text" id="callback-url"
+                 name="callback-url" value="{{githubBaseUrl}}github/callback/{{uuid}}"
+                 readonly
+          />
+        </div>
+
+        <div class="field-group">
+          <label for="setup-url">Setup URL</label>
+          <input class="text full-width-field" type="text" id="setup-url"
+                 name="setup-url" value="{{githubBaseUrl}}github/setup/{{uuid}}"
+                 readonly
+          />
+        </div>
+
+        <div class="field-group">
+          <label for="webhook-url">Webhook URL</label>
+          <input class="text full-width-field" type="text" id="webhook-url"
+                 name="webhook-url" value="{{githubBaseUrl}}github/webhook/{{uuid}}"
+                 readonly
+          />
+        </div>
+
+        <div class="jiraManualAppCreation__formTitle">Secrets</div>
+        <div class="jiraManualAppCreation__formSubTitle">
+          Paste secrets from your GitHub Enterprise Server configuration.
+        </div>
+
+        <div class="field-group">
+          <label for="client-secret">GitHub client secret</label>
+          <input class="text full-width-field" type="password"
+                 data-aui-validation-field required
+                 id="client-secret" name="client-secret" />
+        </div>
+
+        <div class="field-group">
+          <label for="webhook-secret">Webhook secret</label>
+          <input class="text full-width-field" type="password"
+                 data-aui-validation-field required
+                 id="webhook-secret" name="webhook-secret" />
+        </div>
+
+        <div class="field-group">
+          <label>Private key</label>
+          <div class="jiraManualAppCreation__formFileDropArea">
+            <!--
+              Regex pattern for empty string or strings ending with these file extensions.
+              The empty string pattern is added to avoid showing validation pattern error message when this field is empty.
+              File extensions are referenced from https://www.microfocus.com/documentation/visual-cobol/vc60/VS2017/BKCJCJCERTS001.html
+            -->
+            <input class="jiraManualAppCreation__formFileInput" id="private-key"
+                   data-aui-validation-pattern="^\s*$|^.*\.(der|DER|cer|CER|pem|PEM|spc|SPC|p7a|P7A|p7b|P7B|p7c|P7C|p8|P8|pfx|PFX|p12|P12)$"
```

Comment:
Where did this list of file extensions come from? How does this work? From my experience creating GH apps, the extension is always .pem so I don't think most of those are necessary. 

Also, have you tested this? We should render an error message like 'Incorrect file type' or something when a file with the wrong extension is added.

Reply:
I wasn't sure of the different types of files for this. So, searched and found a list of private keys file types here: https://www.microfocus.com/documentation/visual-cobol/vc60/VS2017/BKCJCJCERTS001.html

Are we sure that its always gonna be just `.pem` files? If so, then can change the regex.

And as for the error message, I'm not sure if we need a specific message. The design shows this error message for wrong file types.
![image](https://user-images.githubusercontent.com/13076255/177271712-973bf693-4a59-4b98-bb2a-3ea2af970d28.png)

Reply:
Can I get you to add a new module for this page under `generalPages`? Prior to us figuring out how to add new pages in Jira is made sense to test out new pages by hijacking the render of another page but now that's implemented, we should be adding new pages in connect when they are created.

Reply:
Pretty sure it's only .pem but you could do some investigation into private keys in GH apps to make sure.

Reply:
I'll just update the regex to just `pem`then...